bin/ghe-restore

```diff
@@ -285,6 +285,18 @@ fi
 # Get GHES release version in major.minor format
 RELEASE_VERSION=$(ghe-ssh "$GHE_HOSTNAME" -- 'ghe-config --get core.package-version' | cut -d '.' -f 1,2)
 
+# If the backup being restored is from an appliance with Actions disabled, restoring it onto an appliance with Actions enabled will cause
+# mismatches in the secrets needed for Actions which ultimately results in Actions not working properly.
+ACTIONS_ENABLED_IN_BACKUP=$(git config -f $GHE_RESTORE_SNAPSHOT_PATH/settings.json --list | grep "app.actions.enabled" | xargs)
+# Get the value of app.actions.enabled
+ACTIONS_ENABLED_IN_BACKUP=${ACTIONS_ENABLED_IN_BACKUP#*=}
+# If ACTIONS_ENABLED_IN_BACKUP is still empty, then set it to the default value of "false"
+ACTIONS_ENABLED_IN_BACKUP=${ACTIONS_ENABLED_IN_BACKUP:-"false"}
+if [[ $ACTIONS_ENABLED_IN_BACKUP == "false" ]] && ghe-ssh "$GHE_HOSTNAME" -- 'ghe-config --true app.actions.enabled'; then
```

Comment:
@conorsloan Thanks for the advice on bash, it's looking better now. I don't actually need to replace `.` with `_`, but I had to pipe the result of `grep` to `xargs`, which trims the output (when the keys are not present in `settings.json` grep seems to return whitespaces and carriage returns that break the script unexpectedly.

@boxofyellow  & @tiwarishub: I am now looking into the local backup files to see if we can use a similar approach for this one as we do for ArtifactCache.

Reply:
Btw sorry I didn't spot this earlier, but can't you just do:

``` bash
ACTIONS_ENABLED_IN_BACKUP=$(git config -f <file> app.actions.enabled)
ACTIONS_ENABLED_IN_BACKUP=${ACTIONS_ENABLED_IN_BACKUP:-false}
```

git config supports grabbing the value of a specific path this way

Reply:
I tested this locally and it seemed happy enough:

``` bash
conorsloan@Arcanine ~ % cat /tmp/test.txt
[app "actions"]
	enabled = true
	disabled = false
conorsloan@Arcanine ~ % git config -f /tmp/test.txt app.actions.enabled
true
conorsloan@Arcanine ~ % git config -f /tmp/test.txt app.actions.disabled
false
```

Reply:
Amazing, this works, I wasn't aware of it! I updated it but left the `xargs` for the same reason as earlier (breaks when the value is not in the file)

Reply:
Ah I just spotted this comment! Ignore my comment about this, then. 
It surprises me that this breaks though... is there trailing whitespace in the file itself?